### ISSUE TITLE
back-$ref's in a referenced file no longer resolve in speccy 0.8.x

### ISSUE THREAD

**Issue by devjack** [wework/speccy#185]
If `openapi.yml` $ref's `otherfile.yml`, $ref's in `otherfile.yml` do not resolve.

## Detailed description
To reproduce, I ran speccy 0.7.3 and 0.8.4 over the [same Open API definition](https://github.com/devjack/example-library-specification/blob/master/openapi.yml).

This screenshot shows that speccy 0.7.3 resolves the [$ref from ping.yml to openapi.yml](https://github.com/devjack/example-library-specification/blob/master/endpoints/ping.yml#L15) and 0.8.4 does not (it produces an empty object instead).

<img width="860" alt="screen shot 2018-10-20 at 11 41 08" src="https://user-images.githubusercontent.com/3516066/47250975-4ab21800-d45e-11e8-92e9-1b6fe22d9712.png">

## Context

I define each URL/verb combination as an endpoint and all responses as JSON Schema and use speccy to combine/resolve between the relevant files and formats. An example of this usage is in this [example Open API definition](https://github.com/devjack/example-library-specification) (see the npm scripts for speccy cli usage).

## Example

File: openapi.yml
```yaml
...
paths:
    /ping:
       $ref: "endpoints/ping.yml#/paths/~1ping"

...

components:
    schemas:
        ping:
            $ref: 'schemas/ping.json'
...
```

File: endpoint/ping.yml
```yaml
paths:
    '/ping':
        get:
          responses:
            '200':
              content:
                application/json:
                    schema:
                        $ref: '../openapi.yml#/components/schemas/ping'
```

## Environment

* Node Version: v8.9.4
* OS: MacOS 10.13.4
* Link to your project: Reproduced in this [example Open API definition](https://github.com/devjack/example-library-specification).

**Comment by devjack** [comment]
I've verified that the bug was introduced between 0.7.3 and 0.8.0, likely (?) with the introduction of oas-kit.

**Comment by devjack** [comment]
Raised an issue over at [oas-kit#102](https://github.com/Mermade/oas-kit/issues/102) to address the $ref resolving.

**Comment by philsturgeon** [comment]
@devjack looks like Mike is helping you out over on oas-kit. How's it going?

**Comment by philsturgeon** [comment]
Fixed in #198!
